backend.c

```diff
@@ -283,7 +283,7 @@ static bool fio_io_sync(struct thread_data *td, struct fio_file *f)
 
 requeue:
 	ret = td_io_queue(td, io_u);
-	if (ret < 0) {
+	if ((int)ret < 0) {
```

Comment:
How can this ever be true, it should only return fio_q_status at this point?

Reply:
I was wondering whether there would be any out-of-tree I/O engines for which that code should be kept? If not, I think the code under "if (ret < 0)" can be removed.

Reply:
We version the API anyway, so they would not load. You should bump that version number.